tree/dataframe/src/RLoopManager.cxx

```diff
@@ -822,7 +822,8 @@ void RLoopManager::EvalChildrenCounts()
 
 /// Start the event loop with a different mechanism depending on IMT/no IMT, data source/no data source.
 /// Also perform a few setup and clean-up operations (jit actions if necessary, clear booked actions after the loop...).
-void RLoopManager::Run()
+/// The jitting phase is skipped if the `jit` parameter is `false` (unsafe, use with care).
+void RLoopManager::Run(bool jit)
```

Comment:
Maybe a better approach is making `Jit` a no-op if the global string of code to jit has already been jitted?

Reply:
that's already the case, but in order to know whether the operation is a no-op or not, `Jit` needs to take a global lock :disappointed:

Reply:
Mmm I see, I don't know how much it would cost, it seems it would only be taken once at the beginning (although per unique computation graph, per thread). Maybe another approach is decoupling `Jit` from `RLoopManager`. Then we can call it only once at the beginning of `RunGraphs`, then we can call `RunWithoutJit`.

Reply:
ok, and when the event loop is triggered from a `RResultPtr`, if it's not `RLoopManager::Run` that calls `Jit`, who does? `RResultPtr`?

Reply:
`RLoopManager::Run` can still call `Jit`, but `RunGraphs` calls `RLoopManager::RunWithoutJit`. This would avoid the extra `if` inside of `RLoopManager::Run`.

Reply:
I should probably propose this myself in a subsequent PR

Reply:
mmh I'd rather have a simple `if` than duplicate a method just to omit one line :sweat_smile: